commands/verify.go

```diff
@@ -78,18 +78,20 @@ func verifyDNSSECProofChain(chain *dns.DNSSECProof, query string) (bool, error)
 		return false, errors.New(fmt.Sprintf("failed due to invalid initial_key_tag state."))
 	}
 	// Transition to the ENTERING state
-	fmt.Printf("Number of Zones: %v\n", int(chain.Num_zones))
+	isParentTheSigner, parentSignerIndex := false, -1
 	for i := 0; i < int(chain.Num_zones); i++ {
 		zone := chain.Zones[i]
-		zoneSignerNameBuffer := segments[numSegments-1-i : numSegments]
+		segmentStart := numSegments - 1 - i
+		if isParentTheSigner {
+			segmentStart = numSegments - 1 - parentSignerIndex
+		}
```

Comment:
Will this work if three or more levels in the hierarchy share the same signer? Should be possible, right?

Reply:
Yup, this is because the `parentSignerIndex` will be set during the first non-delegation from parent->child and continues to hold true.

Reply:
Ok, great!